client/src/templates/RouteWithSubRoutes.js

```diff
@@ -1,41 +1,66 @@
 import React from "react";
-import { Route } from "react-router-dom";
+import { Redirect, Route } from "react-router-dom";
 
 import LogoLayout from "./layouts/LogoLayout";
 import NavigationLayout from "./layouts/NavigationLayout";
+import { connect } from "react-redux";
+
+const HOME = "/";
+const LOGIN = "/auth/login";
+const VERIFY_EMAIL = "/auth/verify-email";
 
 // A special wrapper for <Route> that knows how to
 // handle "sub"-routes by passing them in a `routes`
 // prop to the component it renders.
 export const RouteWithSubRoutes = (route) => {
-  const switchLayout = (layout, props) => {
-    switch (layout) {
-      case "logo":
-        return (
-          <LogoLayout
-            {...props}
-            component={route.component}
-            routes={route.routes}
+  const { emailVerified, isAuthenticated, path, props = {} } = route;
+  const { loggedInOnly, notLoggedInOnly } = props;
+
+  return (
+    <Route
+      path={path}
+      render={({ layout, location, ...rest }) => {
+        const Layout = layout === "logo" ? LogoLayout : NavigationLayout;
+        let redirect;
+        // todo: cover all different cases, ensure all the props in the routes config are set correctly
+        //  maybe use LoggedIn, NotLoggedIn, Route components
+        //  or maybe even a state machine to simplify this logic
+        if (loggedInOnly && !isAuthenticated) {
```

Comment:
what's the difference between `loggedInOnly` and `isAuthenticated`

Reply:
yeah I might need to rename things here, 
`loggedInOnly` and `notLoggedInOnly` are props that can be set on a route (which could be also using LoggedIn and NotLoggedIn route components as I've mentioned)
While `isAuthenticated` is the state of the logged in user, this means we did login or signup and got a token.

Reply:
Also, as you might've noticed, I had to clean this up a bit to merge the layout switching and the redirection logic, which uses props from the `routes` config, as well as from the `RouteWithSubRoutes` ocmponent with `render()` method and also wrapped in Redux connect().... this is getting quite complex.
What the next step can be here is to add redirection to "create profile" what @pishili is working on, to make sure if profile is not created it redirects to that page first.
But as you can see that can become very complex and unwieldy logic here so we need to find a way to make it better (maybe state machine...)

Reply:
currently looking if we could use https://statecharts.github.io/ to simplify this whole redirection stuff